### DISCUSSION TITLE
Do I need to connect to other nodes manually in an elixir cluster ?

### DISCUSSION THREAD

**Discussion by sksar** [rabbitmq/khepri#277] [Q&A] [answered]
I have an elixir / phoenix cluster, where I am using Khepri, it is wrapped in a GenServer module, the same GenServer module is executed on all of the nodes, which auto joins to form a cluster, using dns_cluster.

Do, i need to start khepri and then still write the code to connect it explicitly to the other nodes on the same cluster, or khepri does that automatically in the background ?

**Comment by dumbbell** [comment, marked as answer]
Hi!

You will have to explicitly create the Khepri cluster.

I looked at the documentation and source code of `dns_cluster`. If I understand correctly, it connects Erlang nodes as it discovers them, but that’s it. It doesn’t configure other applications like Mnesia. `dns_cluster` doesn’t offer an API currently to run arbitrary code after it discovers a new node either.
